packages/platform-android/src/commands/runAndroid/adb.ts

```diff
@@ -41,6 +41,57 @@ function getDevices(adbPath: string): Array<string> {
   }
 }
 
+function getAllAvailableDevices(adbPath: string) {
+  const emulators = execFileSync(
+    `${process.env.ANDROID_HOME}/emulator/emulator`,
+    ['-list-avds'],
+    {encoding: 'utf8'},
+  )
+    .replace(/\n$/, '')
+    .split('\n')
+    .filter(Boolean)
+    .map(emulator => ({
+      deviceId: undefined,
+      name: emulator,
+      type: 'emulator',
+    }));
+
+  const physicalDevicesList = execFileSync(adbPath, ['devices', '-l'], {
```

Comment:
In other places, we use `execSync('${adbPath} devices')` and so on. No preference here, but would be good to stay consistent.